prac_07/project_management.py

```diff
@@ -0,0 +1,138 @@
+"""
+Estimated time taken: 220 minutes
+Actual time taken: ~300 minutes
+"""
+
+import datetime
+from project import Project
+
+
+def main():
+    projects = []
+    filename = "projects.txt"  # Change this filename if needed
+
+    # Load projects from the data file
+    projects = load_projects(filename)
+
+    choice = ""
+    while choice != "Q":
+        print("- (L)oad projects")
+        print("- (S)ave projects")
+        print("- (D)isplay projects")
+        print("- (F)ilter projects by date")
+        print("- (A)dd new project")
+        print("- (U)pdate project")
+        print("- (Q)uit")
+        choice = input(">>> ").strip().lower()
+
+        if choice == 'l':
+            filename = input("Enter filename to load projects from: ")
+            projects = load_projects(filename)
+
+        elif choice == 's':
+            filename = input("Enter filename to save projects to: ")
+            save_projects(filename, projects)
+
+        elif choice == 'd':
+            display_projects(projects)
+
+        elif choice == 'f':
+            start_date = datetime.datetime.strptime(input("Show projects that start after date (dd/mm/yy): "), "%d/%m/%Y").date()
```

Comment:
perhaps use `filter_date` or some other variable name, instead of `start_date` (as the latter is the same as the existing attribute name)?